utils/initial_data_preparation.py

```diff
@@ -67,22 +75,123 @@ def parse_args():
 
 
 def check_and_clean_data(df):
+    """
+    Processes the provided DataFrame by removing rows containing null values and
+    duplicate rows, updating the DataFrame in place.
+
+    """
     # Checking for null values
     null_counts = df.isnull().sum()
     if np.any(null_counts):
         logger.info("Null values found:")
         logger.info(null_counts[null_counts > 0])
+
+        # Count rows before dropping
+        initial_count = len(df)
+
+        # Drop rows with any null values in place
+        df.dropna(inplace=True)
+
+        # Log how many rows were dropped
+        rows_dropped = initial_count - len(df)
+        logger.info(f"Dropped {rows_dropped} rows containing null values.")
     else:
         logger.info("No null values found.")
 
     # Checking for duplicates
     duplicates = df.duplicated().sum()
     if duplicates:
         logger.info(f"Number of duplicate rows: {duplicates}")
+        # I will keep the first occurrence of each duplicate row and remove the
+        # others:
+        df = df.drop_duplicates(keep="first")
+        logger.info("First occurrence of each duplicate row has been kept. ")
```

Comment:
Very minor: remove the trailing space after the fullstop.

Reply:
This is done.